### PR TITLE
Scanner

### PR DESCRIPTION
**Pull request by cabouezzi**
Implemented Token and Scanner classes

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "dotnet.defaultSolution": "lira.sln"
```
Comment (MarioFares) [inline .vscode/settings.json]: Should be `Lira.sln`
Solution and project names should be uppercase.

File: lira/Scanner.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    private void AddToken(TokenKind kind, object? literal = null)
+    {
+        string text = _source.Substring(_start, _current-_start);
+        Token newToken = new Token(kind, text, _line, literal);
+        _tokens.Add(newToken);
+    }
+
+    #region Character Navigation
+
```
Comment (MarioFares) [inline lira/Scanner.cs:55]: The following 3 functions should be declared as `private`.

File: lira/Token.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace Lira;
+	
+public sealed class Token
+{
+	readonly TokenKind Kind;
```
Comment (MarioFares) [inline lira/Token.cs]: For this and the rest, they should be declared as such:
`public Kind { get; } `

**Review by MarioFares — CHANGES_REQUESTED** [review]
- Solution and project names should be uppercase
- `Advance`, `Peek`, `PeekNext` should be private
- Declare attributes of the `Token` class as public with only `get`

File: lira/Scanner.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public char Advance() => _source[_current++];
```
Comment (MarioFares) [inline lira/Scanner.cs]: The following 3 functions should be declared as private.

File: lira/Token.cs (hunk 3 above)
Comment (cabouezzi) [reply lira/Token.cs]: why not public readonly?

Comment (MarioFares) [reply lira/Token.cs]: @cabouezzi Because it is more idiomatic.

Comment (cabouezzi) [reply lira/Token.cs]: > @cabouezzi Because it is more idiomatic.

you calling me an idiot?

**Review by MarioFares — APPROVED** [review]
Good.

**Pull request merged**